ElementX/Sources/Other/SwiftUI/Views/StackedAvatarsView.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2024 New Vector Ltd.
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+// Please see LICENSE in the repository root for full details.
+//
+
+import SwiftUI
+
+struct StackedAvatarInfo {
+    let url: URL?
+    let name: String?
+    let contentID: String
+}
+
+struct StackedAvatarsView: View {
+    /// Should be negative if we want to get the stacked behaviour
+    let spacing: CGFloat
+    let lineWidth: CGFloat
+    var shouldStackFromLast = false
+    let avatars: [StackedAvatarInfo]
+    let avatarSize: AvatarSize
+    let mediaProvider: MediaProviderProtocol?
+    
+    var body: some View {
+        HStack(spacing: spacing) {
+            ForEach(0..<avatars.count, id: \.self) { index in
```

Comment:
Probably better to make `StackedAvatarInfo` `Identifiable` and then do this so that if we ever animated them, SwiftUI would have a better idea about what was going on.

```suggestion
            ForEach(avatars) { avatar in
```

Reply:
the problem is that I need the index to apply the zStack change but, maybe I could add an .id(contentID) to each one of the elements to solve the issue?